api/src/Service/CampCouponService.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Service;
+
+class CampCouponService {
+    public function __construct(private string $secret) {
+    }
+
+    public function createCoupon() {
+        if ('' == $this->secret) {
+            return 'No Coupon-Key required';
+        }
+
+        return base64_encode(password_hash($this->secret, PASSWORD_BCRYPT, ['cost' => 8]));
```

Comment:
I don't understand the architecture. Which part of this line is the randomness? How does this ever produce different coupon codes?

Reply:
password_hash with BCRYPT results in a string containing a salt and the hashed password it self. the salt is different each time. therefore the string is different each time.
and also therefore you have to use the password_verify function. 
(hashing a password user-input and comparing it with a stored hash does not work)

Reply:
The generated keys are now quite long. I was more expecting a key in the range of 4-8 characters and maybe without the mix of lower/upper case. I assume people would not start spelling them verbally (you never know). But even sending these over WhatsApp/SMS and then enter this key on ecamp (on the mobile) can become quite cumbersome.